src/sentry/static/sentry/app/components/asyncComponent.jsx

```diff
@@ -30,10 +30,19 @@ class AsyncComponent extends React.Component {
   }
 
   componentWillReceiveProps(nextProps, nextContext) {
+    const isRouterInContext = !!this.context.router;
+
+    const currentLocation = isRouterInContext
+      ? this.context.router.location
+      : this.props.location;
```

Comment:
I think `router` can also be a prop, we should just check that too

Reply:
Really? We've never supported that though.

Reply:
Looks like location is always provided so this should be fine